src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -148,8 +150,19 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
         proto = None  # http: or https:
         url = mgr_map.get('services', {}).get('dashboard', None)
         if url:
-            dashboard_urls.append(url.rstrip('/'))
-            p_result = urlparse(url)
+            p_result = urlparse(url.rstrip('/'))
+            hostname = socket.getfqdn(p_result.hostname)
+
+            try:
+                ip = ipaddress.ip_address(hostname)
+            except ValueError:
+                pass
+            else:
+                if ip.version == 6:
+                    hostname = f'[{hostname}]'
+
+            dashboard_urls.append(
+                f'{p_result.scheme}://{hostname}:{p_result.port}{p_result.path}')
```

Comment:
I remember that FQDN first and then the hostnames were (almost) removed from cephadm in favor of IP addresses (due to some issue with the different way docker and podman populated the /etc/hosts), but it now seems we're kind of reverting that move... We still have IPs for containers but FQDNs elsewhere (except in the container addresses):
* This PR
* Alertmanager external URL: https://github.com/ceph/ceph/pull/46218
* FQDN in config files: https://github.com/ceph/ceph/pull/45333

I'm wondering if it'd make sense to recover/reconsider FQDNs for container addresses. Working with some `alertmanager-01.ceph.lab` is always better than `10.95.34.23` (or an IPv6 address). @adk3798 any thoughts on that?

Reply:
Discussed in Dashboard stand-up today: from my side this was more a read-aloud reflection than a change request ;-)

Also we discussed about [conditionally disable the TLS certificate verification in alertmanager](https://prometheus.io/docs/alerting/latest/configuration/#tls_config) as we have already done with the Dashboard to deal with the self-signed certificates from 3rd party components:
```sh
ceph dashboard set-rgw-api-ssl-verify False
ceph dashboard set-grafana-api-ssl-verify False
ceph dashboard set-prometheus-api-ssl-verify False
ceph dashboard set-alertmanager-api-ssl-verify False
```

@s0nea mentioned that in this case there was no need to disable that check from Alertmanager to Grafana, just by ensuring that the certificate matches the FQDN instead of the IP.

Reply:
> Discussed in Dashboard stand-up today: from my side this was more a read-aloud reflection than a change request ;-)
> 
> Also we discussed about [conditionally disable the TLS certificate verification in alertmanager](https://prometheus.io/docs/alerting/latest/configuration/#tls_config) as we have already done with the Dashboard to deal with the self-signed certificates from 3rd party components:
> 
> ```shell
> ceph dashboard set-rgw-api-ssl-verify False
> ceph dashboard set-grafana-api-ssl-verify False
> ceph dashboard set-prometheus-api-ssl-verify False
> ceph dashboard set-alertmanager-api-ssl-verify False
> ```
> 
> @s0nea mentioned that in this case there was no need to disable that check from Alertmanager to Grafana, just by ensuring that the certificate matches the FQDN instead of the IP.

Not sure if it could be done for alertmanager, but for prometheus to avoid disabling LTS I generate a root cert in cephadm and use it to configure prometheus daemon as following. By now I'm not exposing the root cert but we can expose it as config read only parameter for external deployments!

```
  tls_config:
    ca_file: root_cert.pem
```
Candidate WIP PR: https://github.com/ceph/ceph/pull/46400

Reply:
That's a nice thing for the internal component interactions! Unfortunately for Dashboard users it'd be slightly more convoluted: we should somehow expose that root CA cert to users and instruct them to install it in their browsers :-/